src/main/java/com/zendesk/maxwell/row/RowMap.java

```diff
@@ -38,36 +39,39 @@
 	private final LinkedHashMap<String, Object> oldData;
 	private final List<String> pkColumns;
 
+	private static final AtomicInteger rowIdCounter = new AtomicInteger(1);
 	private static final JsonFactory jsonFactory = new JsonFactory();
 
 	private long approximateSize;
 
 	private static final ThreadLocal<ByteArrayOutputStream> byteArrayThreadLocal =
-			new ThreadLocal<ByteArrayOutputStream>(){
-				@Override
-				protected ByteArrayOutputStream initialValue() {
-					return new ByteArrayOutputStream();
-				}
-			};
+		new ThreadLocal<ByteArrayOutputStream>(){
+			@Override
+			protected ByteArrayOutputStream initialValue() {
+				return new ByteArrayOutputStream();
+			}
+		};
 
 	private static final ThreadLocal<JsonGenerator> jsonGeneratorThreadLocal =
-			new ThreadLocal<JsonGenerator>() {
-				@Override
-				protected JsonGenerator initialValue() {
-					JsonGenerator g = null;
-					try {
-						g = jsonFactory.createGenerator(byteArrayThreadLocal.get());
-					} catch (IOException e) {
-						LOGGER.error("error initializing jsonGenerator", e);
-						return null;
-					}
-					g.setRootValueSeparator(null);
-					return g;
+		new ThreadLocal<JsonGenerator>() {
+			@Override
+			protected JsonGenerator initialValue() {
+				JsonGenerator g = null;
+				try {
+					g = jsonFactory.createGenerator(byteArrayThreadLocal.get());
+				} catch (IOException e) {
+					LOGGER.error("error initializing jsonGenerator", e);
+					return null;
 				}
-			};
+				g.setRootValueSeparator(null);
+				return g;
+			}
+		};
 
 	public RowMap(String type, String database, String table, Long timestamp, List<String> pkColumns,
 			BinlogPosition nextPosition) {
+		rowIdCounter.compareAndSet(Integer.MAX_VALUE, 1);
+		this.rowId = rowIdCounter.getAndIncrement();
```

Comment:
Hmm I'm not sure that this method to prevent negative overflow is thread safe.. think I'll make a AtomicRowId class and move the logic for a atomic positive integer there.